### ISSUE TITLE
OneHotEncoder fails on scikit-learn 1.1

### ISSUE THREAD

**Issue by SultanOrazbayev** [dask/dask-ml#930]
By default, dask ml will install latest scikit-learn, however because `_pairwise` has been removed in `1.1`, running `OneHotEncoder` fails.

Steps to reproduce:

Create environment:

```yaml
name: dask_ml
channels:
- conda-forge
dependencies:
- python
- dask-ml
```

Running 

```python
from dask_ml.preprocessing import OneHotEncoder
```

will raise error:

```pytb
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "~/miniconda/envs/dask_ml/lib/python3.10/site-packages/dask_ml/__init__.py", line 4, in <module>
    from dask_ml.model_selection import _normalize
  File "~/miniconda/envs/dask_ml/lib/python3.10/site-packages/dask_ml/model_selection/__init__.py", line 8, in <module>
    from ._search import GridSearchCV, RandomizedSearchCV, check_cv, compute_n_splits
  File "~/miniconda/envs/dask_ml/lib/python3.10/site-packages/dask_ml/model_selection/_search.py", line 18, in <module>
    from sklearn.base import (
ImportError: cannot import name '_is_pairwise' from 'sklearn.base' (~/miniconda/envs/dask_ml/lib/python3.10/site-packages/sklearn/base.py)
```

**Environment**:

- Dask ML version: 2022.1.22
- Scikit-learn version: 1.1.1
- Python version: 3.10
- Operating System: MacOS
- Install method (conda, pip, source): conda

**Comment by TomAugspurger** [comment]
I think https://github.com/dask/dask-ml/pull/921/files is fixing this.

**Comment by TomAugspurger** [comment]
Closed by #921. I'll make a release today.

**Comment by ihalter** [comment]
Same type of issue here, it can't find `sklearn.model_selection._search._check_param_grid`

````
from dask_ml.preprocessing import StandardScaler

ImportError                               Traceback (most recent call last)
Input In [11], in <cell line: 1>()
----> 1 from dask_ml.preprocessing import StandardScaler

File ~\anaconda3\envs\CDA\lib\site-packages\dask_ml\__init__.py:4, in <module>
      1 from pkg_resources import DistributionNotFound, get_distribution
      3 # Ensure we always register tokenizers
----> 4 from dask_ml.model_selection import _normalize
      6 __all__ = []
      8 try:

File ~\anaconda3\envs\CDA\lib\site-packages\dask_ml\model_selection\__init__.py:8, in <module>
      6 from ._hyperband import HyperbandSearchCV
      7 from ._incremental import IncrementalSearchCV, InverseDecaySearchCV
----> 8 from ._search import GridSearchCV, RandomizedSearchCV, check_cv, compute_n_splits
      9 from ._split import KFold, ShuffleSplit, train_test_split
     10 from ._successive_halving import SuccessiveHalvingSearchCV

File ~\anaconda3\envs\CDA\lib\site-packages\dask_ml\model_selection\_search.py:20, in <module>
     18 from sklearn.base import BaseEstimator, MetaEstimatorMixin, clone, is_classifier
     19 from sklearn.exceptions import NotFittedError
---> 20 from sklearn.model_selection._search import BaseSearchCV, _check_param_grid
     21 from sklearn.model_selection._split import (
     22     BaseShuffleSplit,
     23     KFold,
   (...)
     31     _CVIterableWrapper,
     32 )
     33 from sklearn.pipeline import FeatureUnion, Pipeline

ImportError: cannot import name '_check_param_grid' from 'sklearn.model_selection._search' (C:\<redacted>\<redacted>\anaconda3\envs\CDA\lib\site-packages\sklearn\model_selection\_search.py)
````

sklearn: 1.1
python: 3.9.12
dask_ml: 1.9.0
